### ISSUE TITLE
SoundCloud 3.0.5 crashes when started from a Ubuntu Launcher 

### ISSUE THREAD

**Issue by CrazyLemon** [tiliado/nuvolaplayer#302]
So quite complex issue. But first of all..

Info:
Nuvola Player 3 3.0.5
Revision 743bb1dd10654eeee68a8682c72f15ece6d7baf6
WebKitGTK 2.14.5
libsoup 2.54.1
Description:	Ubuntu 16.10
4.8.0-37-generic #39-Ubuntu SMP Thu Jan 26 02:27:07 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux

Ok.. so what i did and what i think it might be the issue.

First start SoundCloud from Dashboard https://help.ubuntu.com/stable/ubuntu-help/unity-dash-intro.html
First issue: double icons, one shown as running other as 'locked'.
![soundcloud_from_dashboard](https://cloud.githubusercontent.com/assets/3321382/23170079/4c6095ca-f84e-11e6-8266-d83bfcbb1813.png)
After 30s the second one ('locked') disappears. Soundcloud works as usual.

Second issue: If i lock Soundcloud that was started from the Dashboard, close it and reopen it - SoundCloud will crash/won't open and the icon from the launcher will disappear. This means that it deletes nuvolaplayer3-soundcloud.desktop file in .local/share/applications/.

Workaround for locking soundcloud to Launcher:
use eu.tiliado.NuvolaAppSoundcloud.desktop from /usr/share/applications/ instead of nuvolaplayer3-soundcloud.desktop
Issue with the workaround: it displays double Soundcloud icons on the Launcher.
Useful info:

```
diff eu.tiliado.NuvolaAppSoundcloud.desktop nuvolaplayer3-soundcloud.desktop 
> 2,9c2,11
> < Name = SoundCloud
> < Exec = nuvolaplayer3 -a soundcloud
> < Type = Application
> < Categories = AudioVideo;Audio;
> < Icon = eu.tiliado.NuvolaAppSoundcloud
> < StartupWMClass = soundcloud
> < StartupNotify = true
> < Terminal = false
> ---
> > Encoding=UTF-8
> > Version=1.0
> > Type=Application
> > Name=SoundCloud
> > Icon=nuvolaplayer3-soundcloud
> > Exec=/usr/lib/nuvolaplayer3/apprunner -a /usr/share/nuvolaplayer3/web_apps/soundcloud
> > StartupNotify=false
> > StartupWMClass=nuvolaplayer3-soundcloud
> > OnlyShowIn=Unity;
> > X-UnityGenerated=true
> 
```

**Comment by fenryxo** [comment]
Hello @CrazyLemon. Quite a complex issue. What happens if you delete all Nuvola desktop launchers from .local/share/applications, then launch SoundCloud from the Dashboard and lock it to the Launcher? Is there any new launcher created?

**Comment by fenryxo** [comment]
I think the cause might be the same as in #301  - Unity cannot match Nuvola/SoundCloud with its new desktop file and creates an invalid launcher entry (`Exec=/usr/lib/nuvolaplayer3/apprunner -a /usr/share/nuvolaplayer3/web_apps/soundcloud` cannot work).

**Comment by CrazyLemon** [comment]
@fenryxo yup. It creates nuvolaplayer3-soundcloud.desktop - content can be seen in the diff above (2nd one - with apprunner). And after i close and reopen it the file disappears.

**Comment by CrazyLemon** [comment]
@fenryxo if you think its the same issue please close whichever one you dislike the most :smile:

**Comment by fenryxo** [comment]
It's the same bug but different symptoms. Better to leave both open.

**Comment by CrazyLemon** [comment]
I have a crash report if it helps in any way to resolve this issue https://paste.sh/G6yNUm_g#G8kfxCVADu7BeHucdmqlVt8K

There's some stuff about .desktop files failing to rescan by gnome-software that might be interesting to you.

**Comment by fenryxo** [comment]
It should be fixed now in Nuvola 3.0.7.

**Comment by CrazyLemon** [comment]
It indeed is fixed. Thanks!
